common/materialContainer/MaterialContainer.cpp

```diff
@@ -123,83 +116,153 @@ Ground Ground::noOre()
   vec.push_back(Material::Sand());
 
   std::string name = "noOre";
-  Ground g(name, vec, .5, 10.0);
-  return g;
+  return std::make_shared<GroundContainer>(name, vec, .5, 10.0);
 }
 
-Ground Ground::soil()
+std::shared_ptr<GroundContainer> GroundContainer::soil()
 {
-  std::vector<Material> vec;
+  std::vector<std::shared_ptr<Material>> vec;
   vec.push_back(Material::Soil());
 
   std::string name = "soil";
-  Ground g(name, vec, .5, 10.0);
-  return g;
+  return std::make_shared<GroundContainer>(name, vec, .5, 10.0);
 }
 
-Ground Ground::rockySoil()
+std::shared_ptr<GroundContainer> GroundContainer::rockySoil()
 {
-  std::vector<Material> vec;
+  std::vector<std::shared_ptr<Material>> vec;
   vec.push_back(Material::Stone());
   vec.push_back(Material::Soil());
   vec.push_back(Material::Gravel());
 
   std::string name = "rockySoil";
-  Ground g(name, vec, .2, 10.0);
-  return g;
+  return std::make_shared<GroundContainer>(name, vec, .2, 10.0);
 }
 
-Ground Ground::sandySoil()
+std::shared_ptr<GroundContainer> GroundContainer::sandySoil()
 {
-  std::vector<Material> vec;
+  std::vector<std::shared_ptr<Material>> vec;
   vec.push_back(Material::Sand());
   vec.push_back(Material::Soil());
 
   std::string name = "sandySoil";
-  Ground g(name, vec, .2, 10.0);
-  return g;
+  return std::make_shared<GroundContainer>(name, vec, .2, 10.0);
 }
 
-Material Ground::getMaterial()
+std::shared_ptr<Material> GroundContainer::getMaterial(int x, int y, int z)
 {
-  double noise = PerlinNoise();
+  double noise = PN(x, y, z);
   return getElement(noise);
 }
 
-Material Material::Coal()
+std::shared_ptr<Material> Material::makenew(std::string name, BlockType type, double freq){
+  materialMap.insert(std::pair<std::string,std::shared_ptr<Material>>(name,std::make_shared<Material>(name,type,freq)));
+  return materialMap[name];
+}
+
```

Comment:
I think you can use `emplace` if you want to avoid typing out those long type declarations. Admittedly, `emplace` reads kinda funny, but I like it.